pkg/api/event/hook_response.go

```diff
@@ -75,10 +78,21 @@ type HookResponse struct {
 	Constraints *Constraints `json:"constraints,omitempty"`
 }
 
+var supportedAMRContraints = []string{model.AMRMFA, model.AMROTP, model.AMRPWD, model.AMRSMS}
+
 type Constraints struct {
 	AMR []string `json:"amr,omitempty"`
 }
 
+func (c *Constraints) Validate() error {
+	for _, amr := range c.AMR {
+		if !slice.ContainsString(supportedAMRContraints, amr) {
+			return fmt.Errorf("unsupported amr constraint %s", amr)
```

Comment:
This error will become server error, as it is constructed with fmt.Errorf.

If you want to warn the developer, you should use our JSON schema validation. The error audience is the developer so in production end user should not see it.